### PR TITLE
Validate payment with Usage History before adding pmt

### PR DESCRIPTION
**Pull request by ptulugu**
This commit fixes the following edge case. Router A is paying B. B reboots and loses all payment history in payment validator. Next time A sends all its txids over, B revalidates pmts that have already been accounted for  leading to double counting. This patch fixes that by looking through entire usage history for the payment txid before adding that payment to debt keeper and usage tracker

### CODE REVIEW

File: rita_common/src/usage_tracker/mod.rs [hunk 1]
```diff
@@ -84,19 +84,27 @@ pub struct FormattedPaymentTx {
...
+    if let Some(txid) = input.txid {
+        FormattedPaymentTx {
             to: input.to,
             from: input.from,
             amount: input.amount,
             txid: format!("{txid:#066x}"),
-        },
-        None => FormattedPaymentTx {
+        }
+    } else if let Some(tx_hash) = input.tx_hash {
```
Comment (jkilpatr) [inline rita_common/src/usage_tracker/mod.rs:94]: why was this change needed?

File: rita_common/src/payment_validator/mod.rs [hunk 2]
```diff
@@ -788,15 +789,17 @@ fn handle_tx_messaging(transaction: TransactionDetails, ts: ToValidate, current_
...
 
-            // update debt keeper with the details of this payment
-            let _ = payment_received(
-                pmt.from,
-                pmt.amount,
-                denom.expect("How did this happen when we already verified existence"),
-            );
+            if !payment_accounted_for(pmt.clone()) {
+                // update debt keeper with the details of this payment
+                let _ = payment_received(
```
Comment (jkilpatr) [inline rita_common/src/payment_validator/mod.rs:794]: call payment accounted for inside update payments, but send it off to the payment validator, part of hte goal of re-sending payment data was to ensure routers didn't enforce on each other, right now the only bug we're trying to fix is duplicate data in the earnings display, which won't be affected by us applying a payment twice due to the way the internals of debt keeper work.

**Review by jkilpatr — CHANGES_REQUESTED** [review]

File: rita_common/src/usage_tracker/mod.rs (hunk 1 above)
Comment (ptulugu) [reply rita_common/src/usage_tracker/mod.rs:94]: In b21 pmt struct can either be txid from eth or tx_hash from althea, guess i missed this change during the adding althea wallets pr

**Review by jkilpatr — APPROVED** [review]
